store/store_download_test.go

```diff
@@ -1040,3 +1041,36 @@ func (s *storeDownloadSuite) TestTransferSpeedMonitoringWriterUnhappy(c *C) {
 	c.Assert(terr, Equals, true)
 	c.Check(w.Err(), ErrorMatches, "download too slow: .* bytes/sec")
 }
+
+func (s *storeDownloadSuite) TestDownloadTimeoutOnHeaders(c *C) {
+	restore := httputil.MockResponseHeaderTimeout(250 * time.Millisecond)
+	defer restore()
+
+	var mockServer *httptest.Server
+
+	quit := make(chan bool)
+	mockServer = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// block the handler, do not send response headers.
+		select {
+		case <-quit:
+		case <-time.After(5 * time.Second):
```

Comment:
/me assumes that we will eventually hit this timeout in the autopkgtest lp builders

Reply:
oh well, you might be right, let's see.

Reply:
If not LP builders then OBS ones :unamused:

Reply:
Fair enough, bumped to 30s.